helix-core/src/main/java/org/apache/helix/tools/ClusterVerifiers/BestPossibleExternalViewVerifier.java

```diff
@@ -456,15 +450,17 @@ public String toString() {
        + (_resources != null ? Arrays.toString(_resources.toArray()) : "") + "])";
   }
 
+  // TODO: to clean up, finalize is deprecated in Java 9
   @Override
   public void finalize() {
+    super.finalize();
     close();
   }
 
-  private class DryrunWagedRebalancer extends org.apache.helix.controller.rebalancer.waged.ReadOnlyWagedRebalancer {
-    public DryrunWagedRebalancer(String metadataStoreAddress, String clusterName,
+  private static class DryrunWagedRebalancer extends ReadOnlyWagedRebalancer implements AutoCloseable {
```

Comment:
Good call on Autoclosable! Just for my understanding why did we make static ?

Reply:
This is to make it a static nested class. Previously, it's non-static, so it has access to the enclosing class members, which it doesn't need (and actually shouldn't). Moreover, non-static class instance implicitly holds a reference to its outer instance, this is something we don't need and can get away too.